plone/app/upgrade/v60/alphas.py

```diff
@@ -264,3 +268,101 @@ def fix_unicode_properties(context):
     portal = getSite()
     portal.reindexObject()
     portal.ZopeFindAndApply(portal, search_sub=1, apply_func=fix_properties)
+
+
+def cleanup_resources_and_bundles_in_registry(context=None):
+    """Fix registry for es6 resources and new resource registry.
+    """
+    registry = getUtility(IRegistry)
+
+    # We need to upgrade staticresources first.
+    # Otherwise the bundles we delete will come back to haunt us
+    context.upgradeProfile("plone.staticresources:default", dest="208")
+
+    # Remove obsolete records from the registry
+    removed_keys = [
+        "plone.resources/",
+        "plone.lessvariables",
+        "plone.resources.configjs",
+        "plone.resources.last_legacy_import",
+        "plone.resources.less-modify",
+        "plone.resources.less-variables",
+        "plone.resources.lessc",
+        "plone.resources.requirejs",
+        "plone.resources.rjs",
+    ]
+    to_delete = []
+    for key in registry.records:
+        for removed_key in removed_keys:
+            if key.startswith(removed_key):
+                to_delete.append(key)
+                logger.debug(u"Removed record {}".format(key))
```

Comment:
I suggest adding a `break` here: the key has been appended to `to_delete`, so we can go to the next `key` instead of trying all other `removed_keys`. Tiny bit faster, and it avoids adding the key to the list twice. With the single upgrade I tried, this corner case did not happen though.

Reply:
Done this myself.